src/System.Linq/src/System/Linq/Contains.cs

```diff
@@ -12,16 +11,34 @@ public static partial class Enumerable
         public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource value)
         {
             ICollection<TSource> collection = source as ICollection<TSource>;
-            if (collection != null) return collection.Contains(value);
+            if (collection != null)
+            {
+                return collection.Contains(value);
```

Comment:
Yes, it's that inconsistency that makes it strange rather than clearly wrong, so while it often strikes me when I see this bit of code, I didn't open an issue. (By the same token, we could make the other overload O(1) if given a `HashSet<T>` after examining the `Comparer` property, but I don't think it would be common enough that it happens to justify the check for it).